application/holder/src/env/fs/index.ts

```diff
@@ -74,27 +76,27 @@ export function getFolders(paths: string[]): string[] | Error {
     }
 }
 
-export function getFileEntities(files: string[]): File[] | Error {
+export async function getFileEntities(files: string[]): Promise<File[] | Error> {
     if (files.length === 0) {
         return [];
     } else {
         try {
-            return files
-                .map((filename: string) => {
-                    const entity: File | Error | undefined = getFileEntity(filename);
-                    if (entity instanceof Error) {
-                        throw entity;
-                    }
-                    return entity;
-                })
-                .filter((f) => f !== undefined) as File[];
+            const entities = [];
+            for(let i = 0; i < files.length; i++) {
+                const entity: File | Error | undefined = await getFileEntity(files[i])
+                if (entity instanceof Error) {
+                    throw entity
+                }
+                entities.push(entity);
+            }
+            return entities.filter((f) => f !== undefined) as File[];
         } catch (e) {
             return new Error(error(e));
         }
     }
 }
```

Comment:
Section `try {} catch {}` there are from the previous solution (sync solution). As soon as now it's becoming `async` code, at least return type should be changed.

We cannot return `Promise<File[] | Error>` error in promise traditionally means rejecting. I think it should be `Promise<File[]>`.

This piece of code 
```
                const entity: File | Error | undefined = await getFileEntity(files[i])
                if (entity instanceof Error) {
                    throw entity
                }
```
would not work, because in case of rejecting (error) of `getFileEntity` it will throw exception, because no catch section.